### PR TITLE
BUG: fix a crash where a string was prematurely converted to pathlib.Path in an internal heuristic function

### LINKED ISSUE #17562: BUG: wrongly assumed Path when parsing html
I see this in one of the astroquery tests (reproduce it with `pytest -P mpc -R`, I may write up an MRE without referring to astroquery later):
 It's clearly caused by 3bf33f68ad220a00384d42478e43955f62454afe, the tests pass with 6.1.7
```

self = <astroquery.mpc.tests.test_mpc_remote.TestMPC object at 0x136b73740>

    def test_get_observations(self):
        # asteroids
        a = mpc.MPC.get_observations(2)
        assert a['number'][0] == 2
    
        a = mpc.MPC.get_observations(12893)
        assert a['number'][0] == 12893
        assert a['desig'][-1] == '1998 QS55'
>       a = mpc.MPC.get_observations("2019 AA")

astroquery/mpc/tests/test_mpc_remote.py:97: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astroquery/utils/class_or_instance.py:25: in f
    return self.fn(obj, *args, **kwds)
astroquery/utils/process_asyncs.py:29: in newmethod
    result = self._parse_result(response, verbose=verbose)
astroquery/mpc/core.py:1212: in _parse_result
    data = ascii.read("\n".join([o['original_record']
../../../.pyenv/versions/3.12.1/lib/python3.12/site-packages/astropy/io/ascii/ui.py:398: in read
    new_kwargs["guess_html"] = _probably_html(table)
../../../.pyenv/versions/3.12.1/lib/python3.12/site-packages/astropy/io/ascii/ui.py:148: in _probably_html
    table_path = Path(table).expanduser()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = PosixPath('~0FR1K19A00A  C2011 01 29.24643 01 18 02.537-02 41 30.21         22.2 wL~3JL8F51\n~0FR1K19A00A  C2011 01 29...47 46 56.60         20.93GV~7ukZG96\n~0FR1K19A00A 1C2024 03 03.20377105 56 18.827+47 46 54.95         20.97GV~7ukZG96')

    def expanduser(self):
        """ Return a new path with expanded ~ and ~user constructs
        (as returned by os.path.expanduser)
        """
        if (not (self.drive or self.root) and
            self._tail and self._tail[0][:1] == '~'):
            homedir = self._flavour.expanduser(self._tail[0])
            if homedir[:1] == "~":
>               raise RuntimeError("Could not determine home directory.")
E               RuntimeError: Could not determine home directory.

../../../.pyenv/versions/3.12.1/lib/python3.12/pathlib.py:1406: RuntimeError
==================================================================== short test summary info ====================================================================
FAILED astroquery/mpc/tests/test_mpc_remote.py::TestMPC::test_get_observations - RuntimeError: Could not determine home directory.

```

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Fixes #17562

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Whoops, just saw that @pllim beat me to it #17564. Well, that makes for any easy review for me 😅

File: docs/changes/io.ascii/17565.bugfix.rst
Comment (pllim) [inline docs/changes/io.ascii/17565.bugfix.rst:1]: Maybe reword this a little so end user can make sense of it? Really, as far as they are concerned, this bug appears when data starts with a tilda.

File: astropy/io/ascii/tests/test_read.py [hunk 1]
```diff
@@ -1497,6 +1497,8 @@ def test_probably_html(home_is_data):
         ],
         (" <! doctype htm > ", " hello world"),
         [[1, 2, 3]],
+        # regression test for https://github.com/astropy/astropy/issues/17562
+        "~itsatrap",  # looks like a Path, but isn't
```
Comment (pllim) [inline astropy/io/ascii/tests/test_read.py:1501]: Can we use the actual string from original bug report?

```
        ("~0FR1K19A00A  C2011 01 29.24643 01 18 02.537-02 41 30.21         22.2 wL~3JL8F51\n"
         "~0FR1K19A00A  C2011 01 29...47 46 56.60         20.93GV~7ukZG96\n"
         "~0FR1K19A00A 1C2024 03 03.20377105 56 18.827+47 46 54.95         20.97GV~7ukZG96")
```

Comment (neutrinoceros) [reply astropy/io/ascii/tests/test_read.py:1501]: I added both !

**Review by pllim — APPROVED** [review]
LGTM. The code now is consistent with what Brigitta approved in the alt PR. Thanks!
